src/System.Memory/tests/Span/Reverse.cs

```diff
@@ -0,0 +1,331 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System.Runtime.CompilerServices;
+using static System.TestHelpers;
+
+namespace System.SpanTests
+{
+    public static partial class SpanTests
+    {
+        [Fact]
+        public static void ReverseEmpty()
+        {
+            var span = Span<byte>.Empty;
+            span.Reverse();
+
+            byte[] actual = { 1, 2, 3, 4 };
+            byte[] expected = { 1, 2, 3, 4 };
+
+            span = actual;
+            span.Slice(2, 0).Reverse();
+            
+            Assert.Equal<byte>(expected, span.ToArray());
+        }
+
+        [Fact]
+        public static void ReverseEmptyWithReference()
+        {
+            var span = Span<string>.Empty;
+            span.Reverse();
+
+            string[] actual = { "a", "b", "c", "d" };
+            string[] expected = { "a", "b", "c", "d" };
+
+            span = actual;
+            span.Slice(2, 0).Reverse();
+
+            Assert.Equal<string>(expected, span.ToArray());
+        }
+
+        [Fact]
+        public static void ReverseByte()
+        {
+            byte[] actual = { 1, 2, 3, 4, 5 };
+            byte[] expected = { 5, 4, 3, 2, 1 };
+            Span<byte> span = actual;
+            span.Reverse();
+            Assert.Equal<byte>(expected, actual);
+        }
+
+        [Fact]
+        public static void ReverseByteLonger()
+        {
+            for(int length = 0; length < byte.MaxValue; length++)
+            {
+                var actual = new byte[length];
+                for (int i = 0; i < length; i++)
+                {
+                    actual[i] = (byte)i;
+                }
+                byte[] expected = new byte[length];
+                Array.Copy(actual, expected, length);
+                Array.Reverse(expected);
+
+                var span = new Span<byte>(actual);
+                span.Reverse();
+                Assert.Equal<byte>(expected, actual);
+            }
+        }
+
```

Comment:
I think having so many different tests is overkill given the implementation.

Reply:
I would guess that these were copied from the Array.Reverse tests and modified to work with Span. If that is the case, then having test parity with Array.Reverse is worthwhile regardless of the simplicity of the implementation.

Reply:
At minimum, I'd remove the big memory tests - we're having trouble keeping the one we have running as it is. Given that System.Memory seems be a dumping ground and has 37,000+ tests already, I'd favor careful monitoring of code coverage and fewer but better targeted tests.

(btw, if Array.Reverse is now using the simple implementation, I think the same argument can be made for those tests.)

Reply:
> I would guess that these were copied from the Array.Reverse tests 

The original `Array.Reverse` had a pretty complex implementation because of it was half-reflection based, with multiple special fast paths implemented in C++ in the runtime.

This one has a very straightforward implementation, so the testing of many different variants is not warranted.

> if Array.Reverse is now using the simple implementation

It does not quite - the non-generic one at least.

Reply:
> I would guess that these were copied from the Array.Reverse tests and modified to work with Span.

They were not taken from the Array.Reverse tests, but I will take a look at them now and try to simplify/reduce the test set. Are there other tests aside from the ones listed below:
https://github.com/dotnet/coreclr/blob/master/tests/src/CoreMangLib/cti/system/array/arrayreverse1.cs
https://github.com/dotnet/coreclr/blob/master/tests/src/CoreMangLib/cti/system/array/arrayreverse2.cs

Reply:
The main Array.Reverse tests live in CoreFX: https://github.com/dotnet/corefx/blob/master/src/System.Runtime/tests/System/ArrayTests.cs

The tests under https://github.com/dotnet/coreclr/blob/master/tests/src/CoreMangLib/ are not great. I hope we will get rid of them: https://github.com/dotnet/coreclr/issues/12782